### PR TITLE
Using service read object to display table

### PR DESCRIPTION
**Pull request by nialldonnellyfh**
Testing Notes:

Deployed to npm.skunkhenry.com as version 2.5.0-221

Using a service: http://testing.ndonnellytest.skunkhenry.com/#services/deqrvgys6gboumo2fvckur76/apps/deqrvg5r2sfblq2ksgbmjy4i/details

1. Target http://testing.ndonnellytest.skunkhenry.com
2. Run fhc services data-sources deqrvgys6gboumo2fvckur76

It should return a table of data sources associated with the service.

### CODE REVIEW

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* A component owner should review your change (we've tagged these above)
* This change should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* Ensure JIRA is up to date (e.g. deployment notes, tester notes, fix version, release notes)
* There should be an associated fhcap PR with this change - please provide a link to this
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

**Comment by fh-qube** [comment]
SonarQube analysis reported 53 issues:
* CRITICAL 6 critical
* MINOR 43 minor
* INFO 4 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L456) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L484) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L163) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* CRITICAL [Restrict what this loop acts on by testing each property.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L190) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AForIn)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L125) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L198) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L459) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L347) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L20) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/2c17963a1e5368da6eb80ec07e5eccf429818d06/lib/common.js#L455) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* ... 43 more

**Comment by nialldonnellyfh** [comment]
@wei-lee @jasonmadigan Time for a code review?

**Comment by wtrocki** [comment]
LGTM

Only small suggestion added.

**Comment by nialldonnellyfh** [comment]
@wtrocki I can't see the suggestion comment in the PR. Am I missing something?

File: lib/cmd/fh3/services.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ function listServiceDataSources(serviceId, cb){
 
     common.doGetApiCall(fhreq.getFeedHenryUrl(), "api/v2/services/" + service.apps[0].guid + "/data_sources", "Error reading Service data sources. ", function(err, dataSources){
       if (ini.get('table') === true) {
-        services.table = common.createTableForDataSources(dataSources || []);
+        services.table = common.createTableForDataSources(dataSources || [], service);
       }
 
       cb(err, dataSources);
```
Comment (wtrocki) [inline lib/cmd/fh3/services.js]: I know that this is not part for this ticket but if we specify invalid guid fhc will return really bad error message.
`fhc ERR! Error - not 2xx status code. (400)
....`

Maybe we can pass here some nicer string instead by using `err.message` or `err.result.message`. 
Showing entire stack is little bit confusing.

Comment (nialldonnellyfh) [inline lib/cmd/fh3/services.js]: Yes, I think error handling in fh-fhc probably needs its own ticket.

**Pull request merged**